preprocessors/openstreetmap/osm_service.py

```diff
@@ -537,15 +537,45 @@ def OSM_preprocessor(processed_OSM_data, POIs, amenity):
                                         else:
                                             nodes[node]["POIs_ID"] = existingid
     # Use Python Sort function
+    processed_OSM_data2 = compute_street_length(processed_OSM_data2)
     processed_OSM_data2 = (
         sorted(
             processed_OSM_data2,
-            key=lambda x: len(
-                x['nodes']),
+            key=lambda x:
+                x['distance'],
             reverse=True))
+
+    # Delete the distance key
+    for obj in range(len(processed_OSM_data2)):
+        processed_OSM_data2[obj].pop('distance', None)
     return processed_OSM_data2
 
 
+def compute_street_length(processed_OSM_data):
+    # Compute the overall path length
+    for obj in range(len(processed_OSM_data)):
```

Comment:
Just a note that you can iterate on content with for loops in python, not just by index